src/m2kcalibration_impl.cpp

```diff
@@ -120,6 +120,11 @@ void M2kCalibrationImpl::setAdcInCalibMode()
 
 	m_m2k_adc->enableChannel(0, true);
 	m_m2k_adc->enableChannel(1, true);
+
+	m_m2k_adc->loadNbKernelBuffers();
+	m_adc_kernel_buffers = m_m2k_adc->getKernelBuffersCount();
+	std::this_thread::sleep_for(std::chrono::milliseconds(10));
```

Comment:
This sleep is somehow useful because otherwise the application will crash. This is caused because of the device that did not have enough time to perform all previous operations, so iio_device_set_kernel_buffers_count will return the code -EBUSY.
Another possible workaround can be something like this:
`try_again:
	try {
		m_m2k_adc->setKernelBuffersCount(1);
	} catch (m2k_exception &e) {
		if (e.iioCode() == -EBUSY) {
			goto try_again;
		} else {
			THROW_M2K_EXCEPTION(e.what(), e.type(), e.iioCode());
		}
	}`

Reply:
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Res-goto
Better ditch the `goto`  statement and use a `for loop` or something else. I think this try while EBUSY approach is better than random sleeping `x`ms. One thing to be careful is that if you trow an exception in the else branch some methods in the calibration have a 'clean_up' part deleting some allocated arrays. One way of fixing that is using smart pointers.

Reply:
I think sleeping is fine.
If calibration is done in another thread, other threads can catch-up. The setting of a kernel buffer with try/catch is convoluted and not obvious. I'd rather just put in a comment, sleep because setting this value is not "instant", or similar.